### ISSUE TITLE
cli: read "paths" from stdin for commands that take them

### ISSUE THREAD

**Issue by nrdxp** [NixOS/nix#7437]
**Is your feature request related to a problem? Please describe.**
If the user wants to pass a list of paths to say `nix path-info` or `nix-build`, they run the risk of recieving a "too many args" OS error.

Since passing derivations around can be very efficient when handled in batches it would be nice not to have a cap on this. The current workaround is `xargs`, which I guess is fine, but not exactly intuitive or necessarily efficient.

**Describe the solution you'd like**
I could see it two ways, if the command recieves a non-empty stdin, simply add it to whatever list of paths already exist in the cli, or we could make it explicit and require something like `nix-build -` to specify reading paths from stdin.

**Describe alternatives you've considered**
Basically just keep using xargs, but even if the arg limit is not an issue, shell pipelines are far more readable than nested process substititions, e.g:
```console
$ which svn | nix-store -qR | nix-store --realize
# is nicer than
$ nix-build $(nix-store -qR $(which svn))
```

**Additional context**
Hit this a couple of times when trying to ensure the entire build closure is cached by parsing the results of a `--dry-run`: [divnix/std-action/run/build.sh#14](https://github.com/divnix/std-action/blob/b01efeb31dca2df9f7672233ac1de92068be3671/run/build.sh#L14)

**Comment by nrdxp** [comment]
Just found `nix-store --export` and it's probably the most important usecase for this, since using `xargs` wouldn't wok in this case to serialise a potentially large list of derivations at once to another store.
